tests/integration/network/mempool_test.go

```diff
@@ -0,0 +1,261 @@
+
+			status, err := tmClient.Status(ctx)
+			s.Require().NoError(err)
+			nextHeight := uint64(status.SyncInfo.LatestBlockHeight) + 1
+
+			// pay for fees of next tx for zeroAccount (account for bid sequence)
+			tx2, err := acc.CreateTx(ctx, num, seq+2, 1000000, 1000000, nextHeight, updateMsg)
+			s.Require().NoError(err)
+
+			seq2, num2, err := getAccount(ctx, authtypes.NewQueryClient(cc), *zeroAccount)
+			s.Require().NoError(err)
+
+			// spends all funds in account on fee deduction -> fees are refilled after bid
+			txToWrap, err := zeroAccount.CreateTx(ctx, num2, seq2, 1000000, 3000000, 3000000, msg)
+			s.Require().NoError(err)
+
+			// first delegate tx (just used to increment sequence)
+			firstDelegateTx, err := zeroAccount.CreateTx(ctx, num2, seq2, 1000000, 1000000, 1000000, msg)
+			s.Require().NoError(err)
+
+			// ordered after bid, and shld fail in PrepareProposal as there will be no funds to pay (will be removed from lane)
+			secondDelegateTx, err := zeroAccount.CreateTx(ctx, num2, seq2+1, 1000000, 1000000, 1000000, msg)
+			s.Require().NoError(err)
+
+			// create a bid wrapping firstDelegateTx, tx2 -> i.e spend funds in zeroAccount and refill
+			bid := auctiontypes.NewMsgAuctionBid(acc.Address(), sdk.NewCoin(sdk.DefaultBondDenom, math.NewInt(1000000)), [][]byte{txToWrap, tx2})
+			bidTx, err := acc.CreateTx(ctx, num, seq+1, 1000000, 1000000, nextHeight, bid)
+			s.Require().NoError(err)
+
+			// broadcast txs
+			resp, err := tmClient.BroadcastTxSync(ctx, bidTx)
+			s.Require().NoError(err)
+			s.Require().Equal(uint32(0), resp.Code)
+
+			resp, err = tmClient.BroadcastTxSync(ctx, firstDelegateTx)
+			s.Require().NoError(err)
+			s.Require().Equal(uint32(0), resp.Code)
+
+			resp, err = tmClient.BroadcastTxSync(ctx, secondDelegateTx)
+			s.Require().NoError(err)
+			s.Require().Equal(uint32(0), resp.Code)
+
+			// wait for commit of bid
+			s.Require().NoError(waitForTxCommit(ctx, tmClient, cmttypes.Tx(bidTx).Hash()))
+
+			// check mempool size
+			txs, err := tmClient.NumUnconfirmedTxs(ctx)
+			s.Require().NoError(err)
+
+			cmtTxs := uint64(txs.Total)
+
+			// check app mempool size
+			appTxDist, err := blockClient.GetTxDistribution(ctx, &blockservicetypes.GetTxDistributionRequest{})
+			s.Require().NoError(err)
+
+			// check parity
+			appTxs := 0
+			for _, tx := range appTxDist.Distribution {
+				appTxs += int(tx)
+			}
+
+			s.Require().Equal(appTxs, int(cmtTxs))
+		})
+	})
+}
+
+func createFreeTx(delegator sdk.AccAddress, validator sdk.ValAddress, amount sdk.Coin) sdk.Msg {
+	return stakingtypes.NewMsgDelegate(delegator.String(), validator.String(), amount)
+}
+
+func getFirstValidator(ctx context.Context, cc stakingtypes.QueryClient) (sdk.ValAddress, error) {
+	resp, err := cc.Validators(ctx, &stakingtypes.QueryValidatorsRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(resp.Validators) == 0 {
+		return nil, nil
+	}
+
+	return sdk.ValAddressFromBech32(resp.Validators[0].OperatorAddress)
+}
```

Comment:
What's the difference between doing this and `s.NetworkSuite.Network.Validators[0].ValAddress`?

Reply:
Did not know we could do that lol